packages/shared/prisma/schema.prisma

```diff
@@ -898,6 +900,12 @@ model JobExecution {
   jobInputTraceId String? @map("job_input_trace_id")
   trace           Trace?  @relation(fields: [jobInputTraceId], references: [id], onDelete: SetNull) // job remains when traces are deleted
 
+  jobInputObservationId String?      @map("job_input_observation_id")
+  observation           Observation? @relation(fields: [jobInputObservationId], references: [id], onDelete: SetNull) // job remains when traces are deleted
+
+  jobInputDatasetItemId String?      @map("job_input_dataset_item_id")
+  datasetItem           DatasetItem? @relation(fields: [jobInputDatasetItemId, projectId], references: [id, projectId], onDelete: Cascade) // job is removed when dataset run items are deleted
```

Comment:
Given that the `DatasetItem` Model is requiring the combined key of `id` and `projectId` to be unique, we are forced to cascade `onDelete` here as we cannot and do not want to set `projectId` to `NULL` (SetNull [definiton](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/referential-actions#setnull): "when the referenced record is deleted, set these fields to NULL").  

Is there any technical issue/limitation with cascading on delete? If not, I'd prefer to keep the implementation as is. Thanks!

Reply:
Why would it set the project_id to null? We have the FK to project_id. If possible, I would keep the execution if we delete a dataset item.

Reply:
My understanding: 
- ideal behaviour: When deleting a `DatasetItem`, the `jobInputDatasetItemId` will be set to NULL for all created execution jobs. 
- approach 1, given `@relation(fields: [jobInputDatasetItemId, projectId]`: set `onDelete` to `SetNull` --> this will never work as `SetNull` will attempt to null the `jobInputDatasetItemId` (fine) AND the `projectId` on the `jobExecution` object 
- approach 2, assume we change `@relation(fields: [jobInputDatasetItemId, projectId]` to `@relation(fields: [jobInputDatasetItemId]`--> this throws an error ceteris paribus as the argument `references` must refer to a unique criterion in the related model. The reason we are seeing this error is because we require the combined key `id, projectId` to be unique on the `DatasetItem`. 

So the trade-off we are facing: 
- Option A: adjust the schema of the `DatasetItem` such that the field `id` is `@unique` in the model `DatasetItem` and unlock approach 2 above 
- Option B: accept cascading the job on delete of the `DatasetItem` and leave the `DatasetItem` in tact

I would prefer not extending the scope of this change further, opting for Option B, and my key question to you is do you see a strong case for Option A over Option B?